File: clients/javascript/src/models/streamReceiveIn.ts
// this file is @generated





export interface StreamReceiveIn {
    batchSize?: number;
consumerGroup: string;
leaseDurationMillis?: number;
topic: string;
}

export const StreamReceiveInSerializer = {
    _fromJsonObject(object: any): StreamReceiveIn {
        return {
            batchSize: object['batch_size'],
            consumerGroup: object['consumer_group'],
            leaseDurationMillis: object['lease_duration_millis'],
            topic: object['topic'],
            };
    },

    _toJsonObject(self: StreamReceiveIn): any {
        return {
            'batch_size': self.batchSize,
            'consumer_group': self.consumerGroup,
            'lease_duration_millis': self.leaseDurationMillis,
            'topic': self.topic,
            };
    }
}

File: clients/rust/src/models/stream_receive_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct StreamReceiveIn {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub batch_size: Option<u16>,

    pub consumer_group: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub lease_duration_millis: Option<u64>,

    pub name: String,

    pub topic: String,
}

impl StreamReceiveIn {
    pub fn new(consumer_group: String, name: String, topic: String) -> Self {
        Self {
            batch_size: None,
            consumer_group,
            lease_duration_millis: None,
            name,
            topic,
        }
    }
}
